### ISSUE TITLE
mergeObjects doesn't check existence of the `left` argument

### ISSUE THREAD

**Issue by nLight** [mozilla-services/react-jsonschema-form#1094]
`mergeObjects` function doesn't check the arguments which leads to an exception.

Example:
```js
const extractedDefaults = {
  dashboard_config_repository: {
    credentials: undefined
  }
};
const formData = {
  dashboard_config_repository: {
    credentials: {
      username: "user",
      password: "pass"
    }
  }
};

mergeObjects(extractedDefaults, formData);
```

https://github.com/mozilla-services/react-jsonschema-form/blob/0c613e48ca2153473294e3340756e235208600f8/src/utils.js#L225

Potential fix:
```js
export function mergeObjects(obj1, obj2, concatArrays = false) {
  // Recursively merge deeply nested objects.
  var acc = Object.assign({}, obj1); // Prevent mutation of source object.
  return Object.keys(obj2).reduce((acc, key) => {
    const left = obj1 && obj1.hasOwnProperty(key) ? obj1[key] : undefined,
      right = obj2 && obj2.hasOwnProperty(key) ? obj2[key] : undefined;
    if (obj1 && obj1.hasOwnProperty(key) && isObject(right)) {
      acc[key] = mergeObjects(left, right, concatArrays);
    } else if (concatArrays && Array.isArray(left) && Array.isArray(right)) {
      acc[key] = left.concat(right);
    } else {
      acc[key] = right;
    }
    return acc;
  }, acc);
}
```

**Comment by nLight** [comment]
Related https://github.com/mozilla-services/react-jsonschema-form/pull/843/files

**Comment by glasserc** [comment]
This should be fixed by #1095, which was released in in v1.0.7. Please let me know if you still have problems with it.
